src/decorator-annotator.ts

```diff
@@ -15,14 +15,23 @@ import {SourceMapper} from './source_map_utils';
 import {assertTypeChecked, TypeTranslator} from './type-translator';
 import {toArray} from './util';
 
+/**
+ * ConstructorParameters are gathered from constructors, so that their type information and
+ * decorators can later be emitted as an annotation.
+ */
+interface ConstructorParameter {
+  type: ts.TypeNode|null;
```

Comment:
Added docs. This is indeed still a somewhat odd code structure, but at least the fields have names now, and can be documented :-)